code/modules/reagents/chemistry/reagents/withdrawl/_addiction.dm

```diff
@@ -0,0 +1,137 @@
+
+///base class for addiction, handles when you become addicted and what the effects of that are. By default you become addicted when you hit a certain threshold, and stop being addicted once you go below another one.
+/datum/addiction
+	///Name of this addiction
+	var/name = "cringe code"
+	///Higher threshold, when you start being addicted
+	var/addiction_gain_threshold = 600
+	///Lower threshold, when you stop being addicted
+	var/addiction_loss_threshold = 400
+	///Messages for each stage of addictions.
+	var/list/withdrawal_stage_messages = list()
+	///Rates at which you lose addiction (in units/second) if you are not on the drug at that time per stage
+	var/addiction_loss_per_stage = list(0.5, 0.5, 1, 1.5)
```

Comment:
This needs to be significantly slowed down to account for the much longer times between withdrawal stages, currently it can't even progress past stage one.

Reply:
mhm